daemon/config/config_unix.go

```diff
@@ -86,3 +93,11 @@ func verifyDefaultIpcMode(mode string) error {
 func (conf *Config) ValidatePlatformConfig() error {
 	return verifyDefaultIpcMode(conf.IpcMode)
 }
+
+// ProcessPoolsConfig applies the default address pools configuration, if present
+func (conf *Config) ProcessPoolsConfig() error {
```

Comment:
we should avoid this method and pass the Pool (if any) as a network option and not change the ipam init code path